CMake/AbseilFindDependencies.cmake

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright 2017 The Abseil Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include(ExternalProject)
+
+
+#
+## pthread
+#
+find_package(Threads REQUIRED)
+
+#
+## CCTZ
+#
+if(NOT ABSL_ENABLE_BUNDLE)
+
+    find_package(CCTZ REQUIRED)
+
+else(NOT ABSL_ENABLE_BUNDLE)
+
+    find_package(CCTZ QUIET)
+    # embedded dep
+    if(NOT CCTZ_FOUND)
+        # if not found, try to bundle it
+        message(STATUS "No External CCTZ found, Use bundle CCTZ package")
+
+        set(CCTZ_BUNDLE_INTERNAL_INSTALLDIR "${CMAKE_CURRENT_BINARY_DIR}/CCTZ_BUNDLE-prefix/install")
+        set(CCTZ_BUNDLE_INTERNAL_SRCDIR "${CMAKE_CURRENT_BINARY_DIR}/CCTZ_BUNDLE-prefix/src/CCTZ_BUNDLE")
+
+        ExternalProject_Add(CCTZ_BUNDLE
+            GIT_REPOSITORY "https://github.com/google/cctz.git"
+            GIT_TAG "master"
+    #        SOURCE_DIR "${CCTZ_BUNDLE_INTERNAL_SRCDIR}"
+            CONFIGURE_COMMAND "true"                            #  pure makefile, no configure
+            BUILD_COMMAND "make" "PREFIX=${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}"
+                "SRC=${CCTZ_BUNDLE_INTERNAL_SRCDIR}/" "-f" "${CCTZ_BUNDLE_INTERNAL_SRCDIR}/Makefile"
+                "CXX=${CMAKE_CXX_COMPILER}" "CC=${CMAKE_C_COMPILER}"
+            INSTALL_DIR "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}"
+            INSTALL_COMMAND "make" "install"
+                "PREFIX=${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}" "SRC=${CCTZ_BUNDLE_INTERNAL_SRCDIR}/" "-f" "${CCTZ_BUNDLE_INTERNAL_SRCDIR}/Makefile"
+                "CXX=${CMAKE_CXX_COMPILER}" "CC=${CMAKE_C_COMPILER}"
+         )
+
+
+        list(APPEND ABSEIL_PRE_DEPENDENCIES CCTZ_BUNDLE)
+
+        set(CCTZ_INCLUDE_DIRS "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}/include")
+        set(CCTZ_LIBRARIES "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}/lib/libcctz.a")
+
+        install(FILES "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}/lib/libcctz.a" DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
+        install(DIRECTORY "${CCTZ_BUNDLE_INTERNAL_INSTALLDIR}/include/cctz" DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR})
+    endif(NOT CCTZ_FOUND)
+endif(NOT ABSL_ENABLE_BUNDLE)
+
+include_directories(${CCTZ_INCLUDE_DIRS})
```

Comment:
Global include_directories is considered bad form in CMake, target_include_directories is preferred.

Reply:
I know it is discouraged. 
include_directories is associated to the scope of the current cmake file, it can be a problem in large projects where a file has a large number of target or when you export targets, it is not the case here. 

I will change that however, both of you dislike it.